docs/codeql/codeql-cli/extractor-options.rst

```diff
@@ -0,0 +1,136 @@
+*   ``codeql database start-tracing --extractor-option java.group1.option2=102 ...``
+
+``--extractor-option`` requires exactly one argument of the form ``extractor_option_name=extractor_option_value``.  ``extractor_option_name`` is the name of the extractor (in this example, ``java``) followed by a period and then the name of the extractor option (in this example, either ``option1`` or ``group1.option2``).  ``extractor_option_value`` is the value being assigned to the extractor option. The value must match the regular expression pattern of the extractor option (if it exists), and it must not contain newline characters.
+
+Using ``--extractor-option`` to assign an extractor option that does not exist is an error.
+
+The CodeQL CLI accepts multiple ``--extractor-option`` options in the same invocation. If you set a ``string`` extractor option multiple times, the last option value overwrites all previous ones. If you set an `array` extractor option multiple times, all option values are concatenated in order.
+
+You can also specify extractor option names without the extractor name. For example::
+
+*   ``codeql database create --extractor-option option1=abc ...``
+*   ``codeql database start-tracing --extractor-option group1.option2=102 ...``
+
+If you do not specify an extractor name, the extractor option settings will apply to all extractors that declare an option with the given name. In the above example, the first command would set the extractor option ``option1`` to ``abc`` for the ``java`` extractor and every extractor that has an option of ``option1``, for example the ``cpp`` extractor, if the ``option1`` extractor option exists for that extractor.
+
+Setting extractor options from files
+------------------------------------
+
+You can also set extractor options through a file. The CodeQL CLI subcommands that accept ``--extractor-option`` also accept ``--extractor-options-file``, which has a required argument of the path to a YAML file (with extension ``.yaml`` or ``.yml``) or a JSON file (with extension ``.json``). For example::
+
+#   ``codeql database create --extractor-options-file options.yml ...``
+#   ``codeql database start-tracing --extractor-options-file options.json ...``
+
+Each option file contains a tree structure of nested maps. At the root is an extractor map key, and beneath it are map keys that correspond to extractor names. Starting at the third level, there are extractor options and option groups.
+
+In JSON::
+
+    {
+         "extractor" : {
+             “java”: {
+                 "option1" : “abc”,
+                 "group1" : {
+                 "option2" : [ 102 ]
+                 }
+             }
+         }
+     }
+
+
+In YAML::
+
+    extractor:
+     java:
+         option1: “abc”
+         group1:
+             option2: [ 102 ]
+
+The value for a ``string`` extractor option must be a string or a number (which will be converted to a string before further processing).
```

Comment:
The following text seems to be formatted in one large paragraph, since there is a lot of information to get across. If there is a way to break it up a bit it might be helpful (I think you intended that based on looking at the wrong file, perhaps its just not formatting correctly). 🙂